library/Test/Hspec/JUnit/Config/Env.hs

```diff
@@ -22,30 +25,44 @@ envJUnitEnabled = (== Just "1") <$> lookupEnv (envPrefix <> "ENABLED")
 -- Variable names align with setter functions from "Test.Hspec.JUnit.Config":
 --
 -- * @JUNIT_OUTPUT_DIRECTORY@ 'setJUnitConfigOutputDirectory'
--- * @JUNIT_OUTPUT_NAME@ 'setJUnitConfigOutputName
+-- * @JUNIT_OUTPUT_NAME@ 'setJUnitConfigOutputName'
 -- * and so on
+--
+-- Environment variable values will have the string @{base}@ replaced with the
+-- basename of the current directory. This can be useful in a monorepository of
+-- multiple packages, for example: @JUNIT_OUTPUT_FILE={base}.xml@
 envJUnitConfig :: IO JUnitConfig
 envJUnitConfig = do
-  modify <-
-    appEndo . foldMap Endo
-      <$> sequence
-        [ lookupEnvOverride "OUTPUT_DIRECTORY" setJUnitConfigOutputDirectory
-        , lookupEnvOverride "OUTPUT_NAME" setJUnitConfigOutputName
-        , lookupEnvOverride "OUTPUT_FILE" setJUnitConfigOutputFile
-        , lookupEnvOverride "SUITE_NAME" $ setJUnitConfigSuiteName . pack
-        , lookupEnvOverride "SOURCE_PATH_PREFIX" setJUnitConfigSourcePathPrefix
-        , lookupEnvOverride "DROP_CONSOLE_FORMATTING" $
+  env <- getEnvironment
+  base <- takeBaseName <$> getCurrentDirectory
+  pure $ readJUnitConfig base env
+
+readJUnitConfig :: FilePath -> [(String, String)] -> JUnitConfig
+readJUnitConfig base env = modify $ defaultJUnitConfig $ pack base
+ where
+  modify =
+    appEndo $
+      foldMap
+        Endo
+        [ readEnv "OUTPUT_DIRECTORY" setJUnitConfigOutputDirectory
+        , readEnv "OUTPUT_NAME" setJUnitConfigOutputName
+        , readEnv "OUTPUT_FILE" setJUnitConfigOutputFile
+        , readEnv "SUITE_NAME" $ setJUnitConfigSuiteName . pack
+        , readEnv "SOURCE_PATH_PREFIX" setJUnitConfigSourcePathPrefix
+        , readEnv "DROP_CONSOLE_FORMATTING" $
             setJUnitConfigDropConsoleFormatting . (== "1")
         ]
 
-  modify . defaultJUnitConfig . pack . takeBaseName <$> getCurrentDirectory
-
-lookupEnvOverride
-  :: String
-  -> (String -> JUnitConfig -> JUnitConfig)
-  -> IO (JUnitConfig -> JUnitConfig)
-lookupEnvOverride name setter =
-  maybe id setter <$> lookupEnv (envPrefix <> name)
+  readEnv name setter =
+    maybe id (setter . replaceBase base) $ lookup (envPrefix <> name) env
 
 envPrefix :: String
 envPrefix = "JUNIT_"
+
+replaceBase :: String -> String -> String
+replaceBase base = go
+ where
+  go = \case
+    [] -> []
+    '{' : 'b' : 'a' : 's' : 'e' : '}' : rest -> base <> go rest
+    c : rest -> c : go rest
```

Comment:
This is fun and all, but how about

```haskell
replaceBase base =
  T.unpack . T.replace "{base}" (T.pack base) . T.pack
```

Reply:
Ha, I had exactly that at first. I dunno why it felt "worse" to do all the packing / unpacking, can't imagine performance is an issue for such small strings. I'll switch it back.